src/parser.rs

```diff
@@ -348,6 +353,40 @@ pub fn parse_u32_as_u64<Endian>(input: EndianBuf<Endian>) -> Result<(EndianBuf<E
     }
 }
 
+/// Convert a `u64` to a `usize` and return it.
+#[doc(hidden)]
+#[inline]
+pub fn u64_to_offset(offset64: u64) -> Result<usize> {
+    let offset = offset64 as usize;
+    if offset as u64 == offset64 {
+        Ok(offset)
+    } else {
+        Err(Error::UnsupportedOffset)
```

Comment:
We should have a test for the error branch. I don't see one in this commit, maybe in a follow up I haven't got to yet?

If not, it can be a follow up, if you'd prefer.